src/coreclr/vm/dacenumerablehash.inl

```diff
@@ -329,6 +380,16 @@ DPTR(VALUE) DacEnumerableHashTable<DAC_ENUM_HASH_ARGS>::BaseFindNextEntryByHash(
         }
     }
 
+    // check for next table must hapen after we looked through the current.
+    VolatileLoadBarrier();
+
+    // in a case if resize is in progress, look in the new table as well.
+    auto nextBuckets = ((DPTR(PTR_VolatileEntry)*)pContext->m_curTable)[1];
```

Comment:
@hoyosjs - I assume this is correct?  (modulo `auto`).  In this case I want to index on the remote side. Is this the right way to do this?

Reply:
```suggestion
    auto nextBuckets = dac_cast<DPTR(PTR_VolatileEntry)>(pContext->m_curTable)[1];
```

Reply:
I made `GetNext` a common helper. With that it is more convenient if m_curTable is `(DPTR(PTR_VolatileEntry)`